src/client/networkclient.cc

```diff
@@ -0,0 +1,410 @@
+      = &NetworkClient::handleConnectedMessage;
+  message_handlers_["proto_error"]
+      = &NetworkClient::handleProtoErrorMessage;
+  message_handlers_["connections_reply"]
+      = &NetworkClient::handleConnectionsMessage;
+  message_handlers_["registry_reply"]
+      = &NetworkClient::handleRegistryReplyMessage;
+  message_handlers_["method_result"]
+      = &NetworkClient::handleMthdResMessage;
+  message_handlers_["method_error"]
+      = &NetworkClient::handleMthdResMessage;
+  message_handlers_["broadcast_result"]
+      = &NetworkClient::handleMthdResMessage;
+  message_handlers_["plugin_trigger_run"]
+      = &NetworkClient::handlePluginTriggerRunMessage;
+  message_handlers_["request_error"]
+      = &NetworkClient::handleNodeTreeRelatedMessage;
+  message_handlers_["get_list_reply"]
+      = &NetworkClient::handleNodeTreeRelatedMessage;
+  message_handlers_["get_data_reply"]
+      = &NetworkClient::handleNodeTreeRelatedMessage;
+  message_handlers_["get_query_reply"]
+      = &NetworkClient::handleNodeTreeRelatedMessage;
+  message_handlers_["query_error"]
+      = &NetworkClient::handleNodeTreeRelatedMessage;
+  message_handlers_["request_ack"]
+      = &NetworkClient::handleNodeTreeRelatedMessage;
+  message_handlers_["get_bindata_reply"]
+      = &NetworkClient::handleNodeTreeRelatedMessage;
+  message_handlers_["connection_error"]
+      = &NetworkClient::handleConnErrorMessage;
+  message_handlers_["plugin_method_run"]
+      = &NetworkClient::handlePluginMethodRunMessage;
+  message_handlers_["plugin_query_get"]
+      = &NetworkClient::handlePluginQueryGetMessage;
+  message_handlers_["plugin_broadcast_run"]
+      = &NetworkClient::handleBroadcastRunMessage;
+  message_handlers_["plugin_handler_unregistered"]
+      = &NetworkClient::handlePluginHandlerUnregisteredMessage;
+}
+
+void NetworkClient::handleNodeTreeRelatedMessage(msg_ptr msg) {
+  nodeTree()->addRemoteNodeTreeRelatedMessage(msg);
+}
+
+void NetworkClient::handleConnectedMessage(msg_ptr msg) {
+  if (connectionStatus() != ConnectionStatus::Connecting) {
+    if (output()) {
+      *output() << "NetworkClient: Very confusing... "
+        "Received \"connected\" message while already connected." << endl;
+    }
+  } else {
+    if (output()) {
+        *output() << "NetworkClient: Received \"connected\" message." << endl;
+    }
+
+    setConnectionStatus(ConnectionStatus::Connected);
+  }
+}
+
+void NetworkClient::handleProtoErrorMessage(msg_ptr msg) {
+  messages::MsgProtoError* mpe
+      = dynamic_cast<messages::MsgProtoError*>(msg.get());
+  if (mpe) {
+    if (output()) {
+      *output()
+          << "Received protocol error message. Aborting connection..." << endl
+          << "    code: " << mpe->err->code.c_str()
+          << "  msg: " << mpe->err->msg.c_str() << endl;
+    }
+
+    disconnect();
+  }
+}
+
+void NetworkClient::handleConnectionsMessage(msg_ptr msg) {
+  // TODO - is this something that client should implement in a subclass?
```

Comment:
Maybe at least log something in those stub methods, so we know that we've received message that we don't handle yet?